care/users/api/viewsets/users.py

```diff
@@ -203,6 +225,17 @@ def destroy(self, request, *args, **kwargs):
         user.save(update_fields=["is_active"])
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    def has_permission(self, user):
+        requesting_user = self.request.user
+        return (
+            requesting_user == user
+            or requesting_user.is_superuser
+            or (
+                requesting_user.state == user.state
```

Comment:
Shouldn't we also check if the user type is a distadmin or a stateadmin ?

Reply:
> Shouldn't we also check if the user type is a distadmin or a stateadmin ?

Nah, in this case the view should be available to all users in the same state/district (i.e. nurse user can view the details of another nurse user).

Reply:
I might be talking out of context here, why is the user only allowed to see users within the same state/district but nothing else ? also why would you check district when you check state? if state is the same then dist check is irrelevant right ?

Reply:
> I might be talking out of context here, why is the user only allowed to see users within the same state/district but nothing else ? also why would you check district when you check state? if state is the same then dist check is irrelevant right ?

So if you were to login using dummynurse1 and compare the results of following:

UserViewSet (/users)

<img width="1284" alt="image" src="https://github.com/user-attachments/assets/59c01613-74c4-43cb-858e-967aea3160ec">

FacilityUsersViewSet (/facility/id/users)

<img width="1284" alt="image" src="https://github.com/user-attachments/assets/cd376eb5-9aeb-4e06-8bc4-24413b50786d">

Latter would include devdistrictadmin as well (linked to the Dummy Facility 40 in the dummy data).

I used the state/district check so that devdistrictadmin's details page would be viewable to dummynurse1.

We could instead filter for accessible facilities here, which UserViewSet uses, but I'd need to know the conditional check to display "more details" button on FE.

```
user.id in Subquery(FacilityUser.objects.filter(
                        facility_id__in=get_accessible_facilities(requesting_user)
                    ).values("user_id")
                )

```

Reply:
@vigneshhari Following up based on the conversation in the call, so state check should suffice for now, right 🤔